src/test/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptorTest.java

```diff
@@ -44,8 +43,7 @@
     @BeforeEach
     void setEnvironment() {
         ReflectionTestUtils.setField(companyNumberValidationInterceptor, "maxIdLength", 50);
-        ReflectionTestUtils.setField(companyNumberValidationInterceptor, "maxCompanyNumberLength", 8);
-        ReflectionTestUtils.setField(companyNumberValidationInterceptor, "companyNumberPattern", "^([a-z]|[a-z][a-z])?\\d{6,8}$");
+        ReflectionTestUtils.setField(companyNumberValidationInterceptor, "companyNumberPattern", "^[A-Za-z0-9]{0,10}$");
```

Comment:
I think if the reg exp has changed we'll need to change this in the web layer too. See company.number.validator.ts. Please test with the web app before changing to confirm...

Reply:
This work should be done on another ticket as we'll be changing and testing the web app and we want some visibility that we are doing this, so should not affect this PR.

Reply:
No shouldn't affect this PR but surely the work is related to the defect so we need another PR for the same JIRA ticket that fixes the web layer too.

Reply:
We have a ticket that is linked to this JIRA ticket.